### PR TITLE
If cannot access any message streams, other streams are tried as defa…

### PR DESCRIPTION
**Pull request by luk-kaminski**
…ult as well, to prevent unnecessary errors.

## Description
Fixes #https://github.com/Graylog2/graylog-plugin-enterprise/issues/4333
/jpd Graylog2/graylog-plugin-enterprise#6245
## Motivation and Context
Users that can access only non-messages streams (like `events` or `failures`) should still be able to use search page, instead of facing errors, even if they do not specify any stream in the UI. In that case non-messages streams are used as default.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/permissions/UserStreams.java [hunk 1]
```diff
@@ -40,10 +40,24 @@ public UserStreams(StreamPermissions streamPermissions, PermittedStreams permitt
         this.permittedStreams = permittedStreams;
     }
 
-    public ImmutableSet<String> loadAll() {
+    public ImmutableSet<String> loadAllMessageStreams() {
         return permittedStreams.load(streamPermissions);
```
Comment (todvora) [inline graylog2-server/src/main/java/org/graylog/plugins/views/search/permissions/UserStreams.java:44]: I'd suggest renaming the PermittedStreams#load to better represent what it does. I know it's not part of this PR, but when we are already touching the code, we could make it clearer.

**Review by todvora — APPROVED** [review]
LGTM, just one minor suggestion

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/permissions/UserStreams.java (hunk 1 above)
Comment (luk-kaminski) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/permissions/UserStreams.java:44]: Good idea, thx!

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/permissions/UserStreams.java [hunk 2]
```diff
@@ -40,8 +40,22 @@ public UserStreams(StreamPermissions streamPermissions, PermittedStreams permitt
...
-        return permittedStreams.load(streamPermissions);
+    public ImmutableSet<String> loadAllMessageStreams() {
+        return permittedStreams.loadAllMessageStreams(streamPermissions);
+    }
+
+    public ImmutableSet<String> loadAllStreams() {
+        return permittedStreams.loadAll(streamPermissions);
+    }
+
+    public ImmutableSet<String> loadStreamsForEmptyStreamsReplacement() {
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog/plugins/views/search/permissions/UserStreams.java:51]:
```suggestion
    public ImmutableSet<String> loadMessageStreamsWithFallback() {
```
How about naming it like this instead?

**Pull request merged**